mureq.py

```diff
@@ -1,32 +1,61 @@
-"""
-mureq is a replacement for python-requests, intended to be vendored
+"""mureq is a replacement for python-requests, intended to be vendored
 in-tree by Linux systems software and other lightweight applications.
 
 mureq is copyright 2021 by its contributors and is released under the
 0BSD ("zero-clause BSD") license.
 """
+
 import contextlib
 import io
 import os.path
 import socket
 import ssl
 import sys
 import urllib.parse
-from http.client import HTTPConnection, HTTPSConnection, HTTPMessage, HTTPException
-
-__version__ = '0.2.0'
-
-__all__ = ['HTTPException', 'TooManyRedirects', 'Response',
-           'yield_response', 'request', 'get', 'post', 'head', 'put', 'patch', 'delete']
-
-DEFAULT_TIMEOUT = 15.0
+from collections.abc import Generator, MutableMapping
+from http.client import (
+    HTTPConnection,
+    HTTPException,
+    HTTPMessage,
+    HTTPResponse,
+    HTTPSConnection,
+)
+from typing import cast
+
+__version__ = "0.3.0"
+
+__all__ = [
+    "HTTPException",
+    "Response",
+    "TooManyRedirects",
+    "delete",
+    "get",
+    "head",
+    "patch",
+    "post",
+    "put",
+    "request",
+    "yield_response",
+]
+
+DEFAULT_TIMEOUT: int = 15
 
 # e.g. "Python 3.8.10"
 DEFAULT_UA = "Python " + sys.version.split()[0]
 
+Headers = MutableMapping[str, str] | HTTPMessage
+# pyrefly: ignore  # not-a-type
+JsonValue = None | bool | int | float | str | list["JsonValue"] | dict[str, "JsonValue"]
```

Comment:
nitpick, doesn't matter: i personally like doing `from __future__ annotations` and then using eg `JsonValue` directly instead of using quotes. do you prefer the quoted version? just curious!